config/e2e/metricbeat.yaml

```diff
@@ -0,0 +1,294 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: metricbeat
+  namespace: {{ .E2ENamespace }}
+  labels:
+    k8s-app: metricbeat
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: metricbeat-daemonset-config
+  namespace: {{ .E2ENamespace }}
+  labels:
+    k8s-app: metricbeat
+data:
+  metricbeat.yml: |-
+    metricbeat.config.modules:
+      # Mounted `metricbeat-daemonset-modules` configmap:
+      path: ${path.config}/modules.d/*.yml
+      # Reload module configs as they change:
+      reload.enabled: false
+
+    metricbeat.autodiscover:
+      providers:
+        - type: kubernetes
+          node: ${NODE_NAME}
+          hints.enabled: true
+    processors:
+      - add_cloud_metadata:
+      - add_fields:
+          target: ''
+          fields:
+            pipeline: {{ .Pipeline }}
+            build_number: {{ .BuildNumber }}
+            provider: {{ .Provider }}
+            clusterName: {{ .ClusterName }}
+            kubernetes_version: {{ .KubernetesVersion }}
+            stack_version: {{ .ElasticStackVersion }}
+            e2e_test_id: {{ .Pipeline }}-{{ .BuildNumber }}-{{ .Provider }}-{{ .ClusterName }}-{{ .KubernetesVersion }}-{{ .ElasticStackVersion }}
+
+    setup.template.overwrite: true
+    setup.template.append_fields:
+    - name: stack_version
+      type: keyword
+    - name: kubernetes_version
+      type: keyword
+    - name: build_number
+      type: keyword
+    - name: pipeline
+      type: keyword
+    - name: e2e_test_id
+      type: keyword
+    - name: provider
+      type: keyword
+    - name: clusterName
+      type: keyword
+
+    output.elasticsearch:
+      hosts: ['https://${ELASTICSEARCH_HOST}:${ELASTICSEARCH_PORT}']
+      username: ${ELASTICSEARCH_USERNAME}
+      password: ${ELASTICSEARCH_PASSWORD}
+      ssl.certificate_authorities:
+      - /mnt/elastic/monitoring-ca.crt
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: metricbeat-daemonset-modules
+  namespace: {{ .E2ENamespace }}
+  labels:
+    k8s-app: metricbeat
+data:
+  kubernetes.yml: |-
```

Comment:
This doesn't need to run as a dset since it's just querying the k8s api, but we can probably leave that for later?

That said, I wonder if we also want to scrape the prom metrics of the operator, I think we might. But that doesn't necessarily need to be in this PR. We can move these to a deployment in the future and also add the prometheus metrics at that point

Reply:
Ah I see later on we are enabling metrics in the operator. I think we also need to add the prometheus module to metricbeat to actually scrape it

Reply:
I _think_ that's happening auto-magically through he auto-discover hints on the operator resource.

Reply:
> This doesn't need to run as a dset since it's just querying the k8s api, but we can probably leave that for later?

I'm not sure, IIRC autodiscover is per node, [this is also how we deploy it in the recipes](https://github.com/elastic/cloud-on-k8s/blob/master/config/recipes/beats/3_metricbeat-kubernetes.yaml#L18).
I agree it is odd for someone who has been used to prometheus.

Reply:
[It can be either](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-autodiscover.html), though it defaults to node local. This drove me batty before I discovered it was the default:

> scope
>    (Optional) Specify at what level autodiscover needs to be done at. scope can either take node or cluster as values. node scope allows discovery of resources in the specified node. cluster scope allows cluster wide discovery. Only pod and node resources can be discovered at node scope.